src/borg/archiver/create_cmd.py

```diff
@@ -370,6 +371,10 @@ def _process_any(self, *, path, parent_fd, name, st, fso, cache, read_special, d
                     self.print_warning("Unknown file type: %s", path)
                     return
             except (BackupError, BackupOSError) as err:
+                if err is BackupOSError:
+                    if err.errno == errno.EACCES or err.errno == errno.EPERM:
```

Comment:
in case we might get more such errnos that must not be retried, it also would be better to use something like `err.errno in (errno.EACCESS, errno.EPERM, )`, so we can just add new errnos at the end of that tuple.

finally, did you consider what the exact conditions leading to EACCES and EPERM are and whether both should not be retried? is this the same on all OSes?

Reply:
According to the [python documentation](https://docs.python.org/3.10/library/errno.html#errno.EACCES), which is a bit lacking in that aspect imo, they corresponded to "linux/include/errno.h", which references t[he posix standard](https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html#tag_15_03) 
This defines EACCES and EPERM the following way:

- [EACCES] Permission denied. An attempt was made to access a file in a way forbidden by its file access permissions.
- [EPERM] Operation not permitted. An attempt was made to perform an operation limited to processes with appropriate privileges or to the owner of a file or other resource.

As I only very rarely use not-linux OSes I don't know exactly to what error conditions these map on other systems

Reply:
OK, both sound like permanent issues retrying won't fix.

@RayyanAnsari any comments about that and windows?

Reply:
btw, on linux (and other unix), this could be tested by backing up a file with `chmod 000` permissions.

Reply:
Looks fine, EACCESS and EPERM both map to PermissionError.
A test for this would be good, `icacls` could be used on Windows to deny all permissions.

Reply:
I will try adding tests next weekend, if someone else wants to work on that before I wont mind

Reply:
I added a test which works on macOS (and likely also other UNIXoid OSes), but might need more work for windows. @RayyanAnsari

Reply:
@RayyanAnsari it executes the `os.chmod(fn, 0o000)` on windows without complaining, but that does not revoke read permissions for owner, so the test fails on win32. can you add correct code for win32?

Reply:
Amended, let's see. Thanks for helping!

Reply:
Works.